contracts/src/systems/trade/contracts/trade_systems.cairo

```diff
@@ -90,6 +90,7 @@ mod trade_systems {
                     Option::Some((
                         resource_type, resource_amount
                     )) => {
+                        assert(*resource_amount != 0, 'maker resource amount is 0');
```

Comment:
not sure we need the contract changes in this PR. Since it will fail anyway from the balance being 0, so this check isn't needed.

Reply:
i think the issue might be that when you call create_order in the UI with 0 resources it's gonna input an empty list, and that will still create the order nothing inside

Reply:
If balance is `> 0` but `resource_amount == 0` the transaction will pass and an order is created at the moment. So we get weird orders like these haha:
<img width="635" alt="image" src="https://github.com/BibliothecaDAO/eternum/assets/76473430/c3447ed8-c20d-4bd7-9f84-31a569dbf65b">

> @edisontim can you check the calldata when resource or lords is 0 ?  

Indeed it is

Reply:
@ponderingdemocritus @edisontim seems to me that assert is needed then, lgtm